test/net/sourceforge/kolmafia/session/YouRobotManagerTest.java

```diff
@@ -0,0 +1,354 @@
+package net.sourceforge.kolmafia.session;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import net.sourceforge.kolmafia.AdventureResult;
+import net.sourceforge.kolmafia.AscensionClass;
+import net.sourceforge.kolmafia.AscensionPath.Path;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.objectpool.SkillPool;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.request.CharPaneRequest;
+import net.sourceforge.kolmafia.request.CharSheetRequest;
+import net.sourceforge.kolmafia.request.EquipmentRequest;
+import net.sourceforge.kolmafia.request.GenericRequest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class YouRobotManagerTest {
+
+  @BeforeAll
+  private static void beforeAll() {
+    // Simulate logging out and back in again.
+    GenericRequest.passwordHash = "";
+    KoLCharacter.reset("");
+    KoLCharacter.reset("you robot manager user");
+    KoLCharacter.setPath(Path.YOU_ROBOT);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @AfterAll
+  private static void afterAll() {
+    Preferences.saveSettingsToFile = true;
+  }
+
+  @BeforeEach
+  private void beforeEach() {
+    Preferences.setInteger("statbotUses", 0);
+    Preferences.setInteger("youRobotTop", 0);
+    Preferences.setInteger("youRobotLeft", 0);
+    Preferences.setInteger("youRobotRight", 0);
+    Preferences.setInteger("youRobotBottom", 0);
+    Preferences.setString("youRobotCPUUpgrades", "");
+    YouRobotManager.reset();
+    KoLCharacter.setAvatar("");
+    ChoiceManager.lastChoice = 0;
+    ChoiceManager.lastChoice = 0;
```

Comment:
duplication, unnecessary?

Reply:
Oops. Supposed to be lastChoice and lastDecision. Will fix.

Reply:
Done.